src/VisualStudio/Core/Impl/Options/OptionLogger.cs

```diff
@@ -14,7 +15,10 @@ internal static class OptionLogger
 
         public static void Log(OptionSet oldOptions, OptionSet newOptions)
         {
-            foreach (var optionKey in newOptions.GetChangedOptions(oldOptions))
+            if (!(newOptions is IInternalOptionSet internalNewOptions))
```

Comment:
Why have this check instead of just making the parameter type `IInternalOptionSet`?

Reply:
Another example. This is internal. 

---
In reply to: [191925967](https://github.com/dotnet/roslyn/pull/27263#discussion_r191925967) [](ancestors = 191925967)